05.wc/lib/wc.rb

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def wc_output
+  opt = OptionParser.new
+  wc_options = { lines: false, words: false, bytes: false }
+  opt.on('-l') { |v| wc_options[:lines] = v }
+  opt.on('-w') { |v| wc_options[:words] = v }
+  opt.on('-c') { |v| wc_options[:bytes] = v }
+  opt.parse!(ARGV)
+
+  counts_by_file = read_files
+  total_by_file = format_file(wc_options, counts_by_file)
+  total_of_files = format_total(wc_options, counts_by_file) if counts_by_file.size > 1
+  [total_by_file, total_of_files].join
+end
+
+def read_files
+  counts = []
+  l = 0
+  w = 0
+  c = 0
+  ARGF.each do |line|
+    l += 1
+    w += line.split.size
+    c += line.bytesize
+    next unless ARGF.file.eof?
+
+    counts << { lines: l, words: w, bytes: c, path: $stdin.tty? ? ARGF.file.path : nil }
+    l = 0
+    w = 0
+    c = 0
+  end
+  counts
+end
+
+def format_file(wc_options, counts)
+  output = []
+  counts.each do |count|
+    wc_options.each_key do |key|
+      output << format_as_tab(count[key]) if selected_option?(wc_options, key)
+    end
+    output << " #{count[:path]}" unless count[:path].nil?
+    output << "\n"
+  end
+  output.join
+end
+
+def selected_option?(params, option)
+  params[option] || params.values.all? { |v| v == false }
+end
+
+def format_total(wc_options, counts)
+  output_total = []
+  wc_options.each_key do |key|
+    output_total << format_as_tab(counts.sum { |count| count[key] }) if selected_option?(wc_options, key)
```

Comment:
```suggestion
    sum = counts.sum { |count| count[key] }
    output_total << format_as_tab(sum) if selected_option?(wc_options, key)
```

一気に喋ろうとせず、息継ぎをしながら話しましょう。
https://x.com/jnchito/status/1326468865103073280

Reply:
ご指摘いただいて、確かに無理矢理な感じがあるなと思いました。
ご提案いただいた内容に変更しました。